GTBitsOfGood/national-npp
# Justin cullen/41 chapter profile

## What is this?
This pull request includes:
- User interface matches Figma design
- Mongoose models and frontend types are updated to reflect page changes
- Existing data is prepopulated when entering the page
- Required fields must be provided to save changes
- Error and info notifications
- Database is updated when new changes are saved


## Setup



## Steps to Test


- In MongoDB, make sure you're logged in and your user profile has an associated chapter attached to it (you can do this by adding a chapter field and assigning an ObjectId for the chapter you want to it
- Navigate to http://localhost:3000/chapter/profile
- - Edit info and refresh to see if changes were saved (a success notification should pop up)
- Test whether required fields work as intended by emptying a required field and pressing "Save Changes"

## Associated Issues

- #41



## Additional Instructions

- We did not implement the phone number validation because Matt said he would create a standard function across both profiles
- We did not implement limiting profile updating to only certain roles because that was not part of the issue
- We did not remove certain schema/type fields like email because they are still used in other parts of the code
- There may be certain fields that are required when they should not be because the Figma did not specify (optional) after them
